torch/csrc/api/src/utils.cpp

```diff
@@ -0,0 +1,15 @@
+#include <torch/utils.h>
+
+#include <ATen/Context.h>
+
+#include <cstdint>
+
+namespace torch {
+void manual_seed(uint64_t seed) {
+  // TODO: Move this to at::Context
+  at::globalContext().defaultGenerator(at::Backend::CPU).manualSeed(seed);
+  if (at::globalContext().hasCUDA()) {
+    at::globalContext().defaultGenerator(at::Backend::CUDA).manualSeedAll(seed);
```

Comment:
This also doesn't match the semantics of the Python bindings. They generally avoid seeding CUDA PRNGs until CUDA is actually initialized. This will trigger the init, which is quite expensive.

Reply:
Ok good to know,I'll address this in another pass